FranciscoPerez/react-challenge-project-jan-2023
# Fp 5 add logout redirect

## Changes
1. Add logout functionality to `Logout` nav button.
2. Dispatch `logoutUser` thunk and redirect to welcome page upon `Logout` button click.

## Purpose
Current `Logout` button simply redirects to `/login` page.

[Issue #5](https://github.com/Shift3/react-challenge-project-jan-2023/issues/5)

## Approach
Upon `Logout` button click, dispatch `logoutUser` thunk to clear auth redux information. Then use the `useHistory` hook to redirect to welcome page. 


## Pre-Testing TODOs
1. **NOTE:** This branch does not contain user login fix from Issue #4

## Testing Steps
1. Navigate to the `/login` page.
2. Enter email and password.
3. Click `Login` button. (see pre-testing notes)
4. Click `Logout` button. 

## Learning
Reviewed redux thunk and dispatch hook examples on the React Redux website.

[Redux Thunks](https://redux.js.org/usage/writing-logic-thunks)

Closes #5
